ArchiSteamFarm/Bot.cs

```diff
@@ -635,6 +685,27 @@ internal sealed class Bot : IDisposable {
 			return await ArchiWebHandler.GetTradeHoldDurationForTrade(tradeID).ConfigureAwait(false);
 		}
 
+		internal async Task<(Dictionary<string, string> Used, Dictionary<string, string> Unused)> GetUsedAndUnusedKeys() {
+			IEnumerable<Task<Dictionary<string, string>>> tasks = new string[] { KeysToRedeemUsedFilePath, KeysToRedeemUnusedFilePath }.Select(file => GetKeysFromFile(file));
+
+			ICollection<Dictionary<string, string>> results;
+
+			switch (Program.GlobalConfig.OptimizationMode) {
+				case GlobalConfig.EOptimizationMode.MinMemoryUsage:
+					results = new List<Dictionary<string, string>>();
+					foreach (Task<Dictionary<string, string>> task in tasks) {
+						results.Add(await task.ConfigureAwait(false));
+					}
+
+					break;
+				default:
+					results = await Task.WhenAll(tasks).ConfigureAwait(false);
+					break;
+			}
+
+			return (results.ElementAt(0), results.ElementAt(1));
```

Comment:
If you have to access by index then change `ICollection` into `IList` and access by index instead.